### ISSUE TITLE
Use of deprecated function call (result_of)

### ISSUE THREAD

**Issue by ITNano** [fmtlib/fmt#1025]
This is just a minor issue but I thought I ought to mention it anyway. The core.h file contains calls to the deprecated function result_of. result_of has been deprecated since the release of C++17 based on [these reasons](https://en.cppreference.com/w/cpp/types/result_of#Notes) and should not be used. std::invoke_result is the preferred method instead nowadays.

`
... fmt/bundled/core.h:215:25: error: 'result_of<fmt::v5::internal::precision_checker<fmt::v5::internal::error_handler> (int)>' is deprecated: warning STL4014: std::result_of and std::result_of_t are deprecated in C++17. They are superseded by std::invoke_result and std::invoke_result_t. You can define _SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning. [clang-diagnostic-deprecated-declarations]
  typedef typename std::result_of<
`

**Comment by DanielaE** [comment]
You are right. In C++20 `std::result_of' will be outright banned. I've created an experimental implementation using `std::invoke_result` when the compiler allows to https://github.com/DanielaE/fmt/commit/c08ee6885b7275615471654b62e9729eba706bab. My impressions from reading everything I could find on the correct usage of `std::invoke_result`, it should be equivalent. Alas, all major compilers (clang, gcc, and msvc) agree on failing in C++17 mode because of errors like "'visit_format_arg': no matching overloaded function found":  [Travis](https://travis-ci.org/DanielaE/fmt/builds/488164726)

@vitaut: do you happen to have an idea about that I am doing wrong?

**Comment by DanielaE** [comment]
Nevermind - I got it ...
